### ISSUE TITLE
Set default User-Agent header in `Case.call`

### ISSUE THREAD

**Issue by Stranger6667** [schemathesis/schemathesis#717]
If it is used in Python tests, then it falls back to the `requests` default user agent. To be consistent with the CLI behavior, we need to set this header inside the `Case.call` method. 

**Implementation notes**
- Use `schemathesis.constants.USER_AGENT` inside `Case.call` / `Case.call_wsgi` methods
- Use `headers.setdefault("User-Agent", USER_AGENT)`
- Pass `headers={"User-Agent": USER_AGENT}` to the `requests.post` call in `schemathesis.specs.graphql.schemas.GraphQLCase.call`

**Comment by divyanshugit** [comment]
Hello, I would love to work on this issue. Can you suggest some ways to get started?

**Comment by Stranger6667** [comment]
Hi @divyanshugit !

Sorry for re-posting the same message from #712 (the steps are the same :)

I suggest following steps from our [contributing guide](https://github.com/schemathesis/schemathesis/blob/master/CONTRIBUTING.rst#submitting-pull-requests) :) Once you set up a local copy with pre-commit enabled, you can implement points from the task description. Let me know if any of these steps require more information!

**Comment by hebertjulio** [comment]
Hi @Stranger6667 ,

Is it correct to set the default agent whenever the headers are none? And when an agent is defined, will it not be overwritten?

**Comment by Stranger6667** [comment]
Hi @hebertjulio!

> Is it correct to set the default agent whenever the headers are none?

Yes, the main side effect here is that if we don't provide our user agent, then the `requests` user agent will be used. I discovered this issue when comparing the CLI & Python tests - with CLI, there was the valid Schemathesis user agent, but with `pytest`, it was something like `python-requests/2.22.0`. It would be nice to have the same behavior in both cases. 

The code to make it work might look like this:

```python
data["headers"] = data["headers"] or {}
data["headers"].setdefault("User-Agent", USER_AGENT)
```

> And when an agent is defined, will it not be overwritten?

Yep, if you explicitly pass the "User-Agent" header to the `case.call()` or via the `-H "User-Agent: custom-agent/1.0"` CLI option, it won't be overwritten. However, we might add a test to verify this behavior.

**Comment by Stranger6667** [comment]
Fixed in #774
